### PR TITLE
Revert string interp

### PR DESCRIPTION
**Pull request by dnfield**
Reverts cause of https://github.com/flutter/flutter/issues/49601

Adds regression test so we can catch it when we go to reland.

### CODE REVIEW

File: dev/integration_tests/release_smoke_test/lib/main.dart [hunk 1]
```diff
@@ -11,9 +14,16 @@ void main() {
   // The test will grep logcat for any errors emitted by Flutter.
   print(text.toDiagnosticsNode());
   print(text.toStringDeep());
+  // regression test for https://github.com/flutter/flutter/issues/49601
+  final List<int> computed = await compute(_utf8encode, 'test', debugLabel: null);
+  print(computed);
```
Comment (dnfield) [inline dev/integration_tests/release_smoke_test/lib/main.dart]: Unfortunately, this is the only way to test release mode code right now.  See https://github.com/flutter/flutter/issues/34505

File: dev/integration_tests/release_smoke_test/lib/main.dart [hunk 2]
```diff
@@ -2,18 +2,29 @@
...
-void main() {
+Future<void> main() async {
   const Text text = Text('Hello, world!', textDirection: TextDirection.ltr);
   // These calls must not result in an error. They behave differently in
   // release mode compared to debug or profile.
   // The test will grep logcat for any errors emitted by Flutter.
   print(text.toDiagnosticsNode());
   print(text.toStringDeep());
+  // regression test for https://github.com/flutter/flutter/issues/49601
+  final List<int> computed = await compute(_utf8encode, 'test', debugLabel: null);
```
Comment (a14n) [inline dev/integration_tests/release_smoke_test/lib/main.dart]: Is `_utf8encode` necessary? `await compute(utf8.encode, 'test', debugLabel: null)`

**Review by a14n — APPROVED** [review]
LGTM

File: dev/integration_tests/release_smoke_test/lib/main.dart (hunk 2 above)
Comment (dnfield) [reply dev/integration_tests/release_smoke_test/lib/main.dart]: Done

Comment (dnfield) [reply dev/integration_tests/release_smoke_test/lib/main.dart]: I guess it is necessary.  Test fails without it this way.

Comment (a14n) [reply dev/integration_tests/release_smoke_test/lib/main.dart]: >I guess it is necessary. Test fails without it this way.

Quite surprising

Comment (a14n) [reply dev/integration_tests/release_smoke_test/lib/main.dart]: Outch. It's really unfortunate. It looks like a bug to me.

```
E/flutter ( 9950): [ERROR:flutter/lib/ui/ui_dart_state.cc(157)] Unhandled Exception: Invalid argument(s): Illegal argument in isolate message : (object is a closure - Function 'encode':.)
```

**Comment by a14n** [comment]
I will try to reland the PR with more null safety.

File: dev/integration_tests/release_smoke_test/lib/main.dart (hunk 2 above)
Comment (dnfield) [reply dev/integration_tests/release_smoke_test/lib/main.dart]: I think it has something to do with the way tear offs are handled by the VM ... @chinmaygarde or @mraleph would probably know

**Comment by a14n** [comment]
Re-landing PR : #49622

File: dev/integration_tests/release_smoke_test/lib/main.dart (hunk 2 above)
Comment (chinmaygarde) [reply dev/integration_tests/release_smoke_test/lib/main.dart]: Is there a reduced test case for the failure?

Comment (dnfield) [reply dev/integration_tests/release_smoke_test/lib/main.dart]: Call `compute` with utf8.encode as the first argument

Comment (mraleph) [reply dev/integration_tests/release_smoke_test/lib/main.dart]: This is a limitation of Dart isolate ports (albeit not a well documented one) - you can only send top-level tear-offs, not instance tear-offs or arbitrary closures through ports.

That is why `_utf8encode` works (it is a top level tear off) and `utf8.encode` fails. 

I have filed https://github.com/dart-lang/sdk/issues/40370 to make executive decision on finally lifting this restriction.

**Pull request merged**